File: Rest-Api-V3/account-and-wallet-private/funding/funding.js
require('dotenv').config();
const axios = require('axios');
const CryptoJS = require("crypto-js");

async function fetchFundingData(marketCode, limit = 100, startTime, endTime) {
  const apiKey = process.env.API_KEY;
  const secretKey = process.env.API_SECRET;
  const ts = new Date().toISOString();
  const nonce = Math.random().toString(36).substring(2);
  const method = "/v3/funding";
  const queryString = `marketCode=${marketCode}&limit=${limit}&startTime=${startTime}&endTime=${endTime}`;
  const msgString = `${ts}\n${nonce}\nGET\napi.ox.fun\n${method}\n${queryString}`;
  const sign = CryptoJS.enc.Base64.stringify(CryptoJS.HmacSHA256(msgString, secretKey));

  try {
    const response = await axios.get(`https://api.ox.fun${method}?${queryString}`, {
      headers: {
        'Content-Type': 'application/json',
        'AccessKey': apiKey,
        'Timestamp': ts,
        'Signature': sign,
        'Nonce': nonce
      }
    });

    if (response.data.success) {
      return response;
    } else {
      console.error('Failed to fetch funding data');
    }
  } catch (error) {
    console.error('Error fetching funding data:', error.response ? error.response.data : error.message);
  }
}

// Example usage
// start time and end time are in milliseconds and must be within 7 days of each other
(async () => {
  const startTime = Date.now() - 24 * 60 * 60 * 1000; // 24 hours in milliseconds
  const endTime = Date.now();
  const response = await fetchFundingData('BTC-USD-SWAP-LIN', 100, startTime, endTime);
  console.log(response);
})();

File: Rest-Api-V3/account-and-wallet-private/funding/funding.py
import os
import time
import hmac
import hashlib
import base64
import requests
from dotenv import load_dotenv

load_dotenv()

def fetch_funding_data(market_code, limit=100, start_time=None, end_time=None):
    api_key = os.getenv('API_KEY')
    secret_key = os.getenv('API_SECRET').encode('utf-8')
    ts = time.strftime('%Y-%m-%dT%H:%M:%S', time.gmtime())
    nonce = str(int(time.time() * 1000))
    method = "/v3/funding"
    api_url = "api.ox.fun"
    
    query_string = f"marketCode={market_code}&limit={limit}&startTime={start_time}&endTime={end_time}"
    msg_string = f"{ts}\n{nonce}\nGET\n{api_url}\n{method}\n{query_string}"
    
    sign = base64.b64encode(hmac.new(secret_key, msg_string.encode('utf-8'), hashlib.sha256).digest()).decode('utf-8')
    
    headers = {
        'Content-Type': 'application/json',
        'AccessKey': api_key,
        'Timestamp': ts,
        'Signature': sign,
        'Nonce': nonce
    }
    
    try:
        response = requests.get(f"https://{api_url}{method}?{query_string}", headers=headers)
        response.raise_for_status()
        return response.json()
    except requests.exceptions.RequestException as error:
        print('Error fetching funding data:', error)

# Example usage
# start time and end time are in milliseconds and must be within 7 days of each other
start_time = int(time.time() * 1000) - 24 * 60 * 60 * 1000  # 24 hours ago in milliseconds
end_time = int(time.time() * 1000)
response = fetch_funding_data('BTC-USD-SWAP-LIN', 100, start_time, end_time)
print(response)